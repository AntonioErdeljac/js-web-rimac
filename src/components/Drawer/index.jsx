import PropTypes from 'prop-types';
import React, { memo } from 'react';
import Sidebar from 'react-sidebar';

import { Content } from './components';

import './index.scss';

const Drawer = ({ children }) => {
  return (
    <Sidebar sidebar={<Content />} docked sidebarClassName="r-sidebar">
      {children}
    </Sidebar>
  );
};

Drawer.propTypes = {
  children: PropTypes.node.isRequired,
};

export default memo(Drawer);
