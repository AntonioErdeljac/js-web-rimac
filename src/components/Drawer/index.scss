@import '../../scss/bootstrap';

.r-sidebar {
  background-color: $white;
}
