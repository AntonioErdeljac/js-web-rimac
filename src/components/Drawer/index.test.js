import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Drawer from './index';

jest.mock('../../providers', () => ({
  useDrawer: () => ({
    toggle: () => {},
    isOpen: false,
  }),
}));

const makeProps = ({ children } = {}) => ({
  children: children || <p>Content</p>,
});

const renderComponent = (props) => shallow(<Drawer {...props} />);

describe('Drawer Component', () => {
  it('renders correctly according to snapshot', () => {
    expect(toJson(renderComponent(makeProps()))).toMatchSnapshot();
  });

  it('renders proper children', () => {
    const expectedChildren = 'Lorem';

    const wrapper = renderComponent(makeProps({ children: expectedChildren }));
    expect(wrapper.children().debug()).toEqual(expectedChildren);
  });
});
