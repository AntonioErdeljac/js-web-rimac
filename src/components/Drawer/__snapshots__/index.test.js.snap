// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Component renders correctly according to snapshot 1`] = `
<Sidebar
  defaultSidebarWidth={0}
  docked={false}
  dragToggleDistance={30}
  onSetOpen={[Function]}
  open={false}
  pullRight={false}
  shadow={true}
  sidebar={<Memo(Content) />}
  sidebarClassName="r-sidebar"
  styles={Object {}}
  touch={true}
  touchHandleWidth={20}
  transitions={true}
>
  <p>
    Content
  </p>
</Sidebar>
`;
